app/models/questioning.rb

```diff
@@ -66,6 +68,11 @@ def repeatable?
     ancestry_depth > 1 && parent.repeatable?
   end
 
+  # all questionings that can be referred to by a condition
+  def refable_qings
+    previous.reject{|qing| NON_REFABLE_TYPES.include?(qing.qtype_name)}
```

Comment:
We'd speed this up about 7x by making `NON_REFABLE_TYPES` a hash (maybe just add `.index_by(&:itself)` to the end of the declaration) and then use `has_key?` instead of `include?`. 

Can you also plz fix the spacing with the braces.

Reply:
Will fix the spacing, but is the loss of code clarity worth the theoretical speed gains of changing to a hash when `n` is less than 10 on that array comparison? If this call is slow, it's because `previous` is doing its exclusions in ruby instead of being able to query the previous records at the database level.

Reply:
I was just thinking there could be like 50 previous questions, so it's either 50 hash lookups or 50*7 array comparisons. I don't think code clarity is all that sacrificed. Maybe we need a data structure .... OH! This is exactly what Set is for. Let's use that! You can do `.to_set` at the end, then the `.includes?` will be O(1) instead of O(n).

Reply:
Using `set` here sounds good.